### PR TITLE
Remove setup_requires from setup.cfg

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Remove `setup_requires` from `setup.cfg`.

`setup_requires` worked in the original `setup.py`-only version but I have never found a version of `setuptools` where it was supported in `setup.cfg`. But since it was possible that there was some old version of `setuptools` that did indeed support this, I kept it around for v3. However I really do not think it is needed at this point, and this reduces the duplication of build requirements specifications. Now they should only go in `pyproject.toml` and `build-constraints.txt`. Install requirements remain in `install_requires` and `requirements.txt` as before.

### Types of change

Chore.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by svlandeg** [comment]
Will this disable some use-cases or users installing/compiling via `setup.py`?

**Comment by adrianeboyd** [comment]
No, not really. Since I redid all the packages in Nov. 2020, users calling `python setup.py` have needed to install the build requirements separately first because `setup.py` imports from cython directly. If someone did still happen to be reading in the requirements from `setup.cfg`, they can switch to reading in the exact same requirements from `pyproject.toml` now.

The intention was for `setup_requires` to be deprecated and it got very close to being actually deprecated last year, although I see that they reverted this decision to some degree for `setup.py` (related to dynamic build-time requirements, which doesn't affect us).

The history of `setuptools` is pretty hard to follow, but when I worked on this in 2020 I never found a recent version of `setuptools` that supported `setup_requires` from `setup.cfg` at all. The exact same setup in `setup.py` worked as intended (with our pre-`pyproject.toml` setups), but not from `setup.cfg`. Since we were switching to `pyproject.toml`, I didn't spend a lot of time trying to track it down further.
